runtime/complex.go

```diff
@@ -140,3 +161,42 @@ func complexHash(f *Frame, o *Object) (*Object, *BaseException) {
 	}
 	return NewInt(hashCombined).ToObject(), nil
 }
+
+func complexArithmeticOp(f *Frame, method string, v, w *Object, fun1 func(v, w complex128) complex128, fun2 func(v complex128, w float64) complex128) (*Object, *BaseException) {
+	if w.isInstance(ComplexType) {
+		return NewComplex(fun1(toComplexUnsafe(v).Value(), toComplexUnsafe(w).Value())).ToObject(), nil
+	}
+
+	floatW, ok := floatCoerce(w)
+	if !ok {
+		if math.IsInf(floatW, 0) {
+			return nil, f.RaiseType(OverflowErrorType, "long int too large to convert to float")
+		}
+		return NotImplemented, nil
+	}
+	return NewComplex(fun2(toComplexUnsafe(v).Value(), floatW)).ToObject(), nil
```

Comment:
I would just convert floatW in this case to a complex(floatW, 0) and call fun1 in all cases. Then you can get rid of the fun2 param altogether.